restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreter.java

```diff
@@ -975,17 +975,17 @@ private void onTagMessage(Object message) throws TransitionFailedException, Tran
                 fsm.transition(message, startDialing);
             } else if (Verbs.fax.equals(verb.name())) {
                 fsm.transition(message, caching);
-            } else if (Verbs.play.equals(verb.name())) {
+            } else if (Verbs.play.equals(verb.name()) && callState != CallStateChanged.State.COMPLETED) {
```

Comment:
it seems we are preventing Verb execution in COMPLETED state sparringly (not all verbs have this protection), and out of FSM transition logic.
Why dont all the verbs have same condition?
Could we prevent same thing with actual FSM configuration (adding/removign transition)?

Reply:
so, the issue is that this decision is not based on VI own state, but on the parituclar call state, hence the usage of if statements rather than FSM transitions, right?

Reply:
Would be nice to modularize which verbs are meant to be executed if call.COMPLETED or not?
maybe a private method supported by a Map, if the verb is container in the MAP proceed,if not,prevent. The name of the method may help to understand what is going on

suggestion : "isExecutionAllowedWhenCompleted"

the private method may introduce some debug logs to indicate if teh verb was allowed to be executed or not in addition.

Reply:
@jaimecasero you answered correctly all the questions and you expressed the same concerns I had.

Some verbs such as SMS, Fax, Email, can be executed even when the call is completed but other not.

Also, I was also thinking that this is not an elegant way to control which verb should be executed depending on the Call state but due to the urgency of the issue and the fact that FSM refactoring is on the plan, I decided to go the fast way and wait for the CI results.